### PR TITLE
Add documentation for publishing snapshots to maven

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
Add documentation for publishing snapshots to maven

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: WORKFLOWS.md [hunk 1]
```diff
@@ -9,6 +9,9 @@
     - [Integration](#integration-2)
   - [Auto Github Releases](#auto-github-releases)
     - [Integration](#integration-3)
+  - [Publishing snapshots to Maven](#publishing-snapshots-to-maven)
```
Comment (dblock) [inline WORKFLOWS.md:12]: Capitalize Snapshots and let's say "Publishing Java Plugins Snapshots"?

File: WORKFLOWS.md [hunk 2]
```diff
@@ -80,4 +83,71 @@ OpenSearch project releases add a [github tag](https://docs.github.com/en/rest/g
 
 1. Setup release notes directory with the repo, see OpenSearch [example](https://github.com/opensearch-project/OpenSearch/tree/main/release-notes). This workflow is dependent on predictable release notes based on a release tag.
 2. Setup the workflow in your repo similar to [auto-release.yml](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/auto-release.yml) in OpenSearch.
-3. Backport this workflow to all release branches i.e 2.x, 1.x etc.
\ No newline at end of file
+3. Backport this workflow to all release branches i.e 2.x, 1.x etc.
+
+## Publishing snapshots to Maven
+
+OpenSearch project uses GitHub Actions to publish snapshots to maven.
```
Comment (dblock) [inline WORKFLOWS.md:90]: Let's link https://mvnrepository.com/artifact/org.opensearch?

File: WORKFLOWS.md [hunk 3]
```diff
@@ -80,4 +83,71 @@ OpenSearch project releases add a [github tag](https://docs.github.com/en/rest/g
...
-3. Backport this workflow to all release branches i.e 2.x, 1.x etc.
\ No newline at end of file
+3. Backport this workflow to all release branches i.e 2.x, 1.x etc.
+
+## Publishing snapshots to Maven
+
+OpenSearch project uses GitHub Actions to publish snapshots to maven. 
+
+### Configuration
+In order to on-board, follow the below steps:
```
Comment (dblock) [inline WORKFLOWS.md:93]: onboard is 1 word ;)

**Review by dblock — CHANGES_REQUESTED** [review]

File: WORKFLOWS.md (hunk 2 above)
Comment (gaiksaya) [reply WORKFLOWS.md:90]: Any idea what is the snapshot equivalent for that? I know https://aws.oss.sonatype.org/content/repositories/snapshots/

File: WORKFLOWS.md [hunk 4]
```diff
@@ -80,4 +83,71 @@ OpenSearch project releases add a [github tag](https://docs.github.com/en/rest/g
 
 1. Setup release notes directory with the repo, see OpenSearch [example](https://github.com/opensearch-project/OpenSearch/tree/main/release-notes). This workflow is dependent on predictable release notes based on a release tag.
 2. Setup the workflow in your repo similar to [auto-release.yml](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/auto-release.yml) in OpenSearch.
-3. Backport this workflow to all release branches i.e 2.x, 1.x etc.
\ No newline at end of file
+3. Backport this workflow to all release branches i.e 2.x, 1.x etc.
+
+## Publishing Java Plugins Snapshots
+
+OpenSearch project uses GitHub Actions to publish snapshots to maven.
```
Comment (dblock) [inline WORKFLOWS.md:90]: Maven is capitalized I believe.

File: WORKFLOWS.md [hunk 5]
```diff
@@ -80,4 +83,71 @@ OpenSearch project releases add a [github tag](https://docs.github.com/en/rest/g
...
+In order to test the maven snapshot publication, update the Snapshots target repository URL from maven to a local file system path in your build.gradle file.
+
+```
+  maven {
+      name = "Snapshots" //  optional target repository name
+      url = "snapshots"
+  }
+  
+```
+Run the command ./gradlew \<taskName> which will publish the files to the local file system under the 'snapshots' folder, then scan with `tree` or your favor directory scanner to see the content along with sha and md5 files.
```
Comment (dblock) [inline WORKFLOWS.md:122]: Let's quote `./gradlew...` so it appears as code.

**Review by dblock — COMMENTED** [review]
Looks good, just nits on formatting to keep things clean and let's merge it.

**Review by saratvemulapalli — APPROVED** [review]

**Pull request merged**